enhancements/insights/config-api.md

```diff
@@ -0,0 +1,245 @@
+
+### Goals
+
+The goal of this enhancement is to propose a basic structure for a new Insights configuration API and define new configuration options. This includes following:
+
+- Define a new configuration option providing a way to trigger on demand Insights gathering
+- Define a new configuration option providing a way to include/exclude particular Insights data gatherers
+- Provide Insights status & reporting in the status field in the new API
+
+### Non-Goals
+
+- Move the majority of the current configuration options to a new configuration API - Insights has quite a lot of configuration options
+  and it does not make sense to expose everything in the new configuration API 
+- Removing the original configuration via `support` secret
+
+## Proposal
+
+The proposal is to introduce a new Insights `CustomResourceDefinition` in the OpenShift API providing a way to configure the Insights operator and to report its status. The new CRD should follow a common CRD structure providing `spec` and `status` attributes. 
+
+```go
+type InsightsSpec struct {
+	// Gathering spec attribute includes all the configuration options related to
+	// gathering of the Insights archive and its uploading to the ingress.
+	GatheringConfig GatheringConfig `json:"gatheringConfig,omitempty"`
+}
+
+type InsightsStatus struct {
+	// GatheringStatus provides basic information about the last Insights gathering.
+	GatheringStatus GatheringStatus `json:"gatheringStatus,omitempty"`
+	// SimpleContentAccessStatus provides basic information about the SCA controller.
+	SimpleContentAccessStatus SimpleContentAccessStatus `json:"simpleContentAccessStatus,omitempty"`
+	// ClusterTransferStatus provides basic information about the ClusterTransfer controller.
+	ClusterTransferStatus ClusterTransferStatus `json:"clusterTransferStatus,omitempty"`
+	// ReportStatus provides general Insights analysis results.
+	ReportStatus ReportStatus `json:"reportStatus,omitempty"`
+}
+
+type GatheringConfig struct {
+	// EnableObfuscation is flag enabling the global obfuscation of the IP addresses and base domain
+	// in the Insights archive data.
+	EnableObfuscation bool `json:"enableObfuscation,omitempty"`
+	// Enables user to force Insights data gathering by setting a new reason. Gathering can be forced even
+	// when all the gatherers are disabled.
+	ForceGatherReason string `json:"forceGatherReason,omitempty"`
+	// List of gatherers to be excluded from the gathering. All the gatherers can disabled by providing "all" value.
+	// If all the gatherers are disabled, the Insights operator does not gather any data.
+	DisabledGatherers []string `json:"disabledGatherers,omitempty"`
```

Comment:
Your user stories are currently bullet points.  Can we make them subsections where we discuss the config that they'd use to accomplish their goals?  Because it feels like there's a use-case for:

1. I don't want the gather polling, so I'll set `disabledGatherers: ["all"]`.
2. Oh, no!  Terrible bug.  Better fire off a one-off gather to send this up to my helpful support contact...  Uhh, I guess I...
    1. Clear `disabledGatherers`.
    2. Touch `forceGatherReason`.  Racing against the poller that the previous step just re-enabled.
    3. Watch `status.lastGatheringTime`... ah, there's a bump.  I guess that means the upload completed successfully too?
    4. Set `disabledGatheres: ["all"]` again.  Or maybe this could get turned off after the gather was launched without waiting for the gather to complete?  Not clear how folks would know when "gather was launched" happened...

Personally, it feels like things would be cleaner if there was a separate knob for "please poll the gatherer", so folks could poke `forceGatherReason` and walk away instead of having to fiddle with `disabledGatherers`.  That would also mean you wouldn't need the `all` special-case value handling in `disabledGatherers`.

Reply:
Yes I think it's a good idea. I think the scenario you described will be hopefully more simple but let me update the section.

Reply:
I mean:

```go
type GatheringConfig struct {
  ...
  // enablePolling enables periodic gathering.  Gathering may also be triggered manually via forceGatherReason, regardless of the enablePolling value.
  EnablePolling bool `json:"enablePolling"`
  ...
}
```

or similar, so folks can disable periodic gathering without having to tweak `disabledGatherers` (separating "what gets gathered?" from "when does gathering happen?").

Reply:
I see. Thanks. Well I think I originally had something very similar and it was discussed in https://github.com/openshift/enhancements/pull/1037#discussion_r815831988. I think there's an effort to have as few "knobs" as possible.